### PR TITLE
[VBLOCKS-2891] Getting started docs

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] ~Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code~
 - [x] ~Tested code changes and observed expected behavior in the example app~
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

This PR adds new documentation to help users get up and started quicker and with less friction.

## Breakdown

- Getting Started docs.

## Validation

- N/A

## Additional Notes

N/A

### CODE REVIEW

File: COMMON_ISSUES.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Common Issues
 
-* Running the example app on Android using `yarn run android` may fail if the emulator is still starting up. When this happens, you can re-run the app once the emulator is fully started.
+* Expo is not an officially supported use case.
 
-* Running the example app on iOS using `yarn run ios` may fail if the target simulator is not running or a different simulator instance is already running. Usually, quiting the simulator and running `yarn run ios` again will fix the issue.
+* Running the example app on Android using `yarn run android` may fail if the emulator is still starting up. When this happens, you can re-run the app once the emulator is fully started.
 
-* Interacting with notifications on Android will not foreground the application.
+* Twilio Programmable Voice will not and cannot work on an iOS simulator. This is a limitation imposed by Apple.
```
Comment (bobiechen-twilio) [inline COMMON_ISSUES.md:7]:
```suggestion
* The Twilio Voice React Native SDK is integrated with the iOS CallKit framework to provide the best call progress and audio experience, and requires running on actual devices.
```

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  ...
+}
+```
+
+The following lifecycle methods need to be hooked:
+  - `onCreate`
+  - `onTerminate`
+
+## iOS
+To get started on iOS, please note that the Twilio Voice React Native SDK will not work on an iOS simulator. This is a limitation imposed by Apple and there is no way to circumvent it.
```
Comment (bobiechen-twilio) [inline docs/getting-started.md:140]: Similar suggestion as the one in common issues

**Review by afalls-twilio — APPROVED** [review]
LGTM

**Review by bobiechen-twilio — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -13,6 +13,8 @@ Please check out the following if you are new to Twilio's Programmable Voice or
 - [Programmable Voice](https://www.twilio.com/docs/voice/sdks)
 - [React Native](https://reactnative.dev/docs/getting-started)
 
+Please note that we do not officially support Expo. Our officially supported use case is a bare React Native application that targets iOS and Android. When following the React Native environment setup guide, please select "React Native CLI".
```
Comment (charliesantos) [inline README.md:16]: This sounds like we're not open to supporting Expo in the future.  I don't think this message even belongs here. The  "please select "React Native CLI"" should be part of a setup instructions. The expo part, we don't have to mention here. Or we can add it in the common issues if it's not there yet.

**Review by charliesantos — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (mhuynh5757) [reply README.md:16]: It's also part of the common issues.

**Review by charliesantos — APPROVED** [review]

**Pull request merged**